src/mon/OSDMonitor.cc

```diff
@@ -1663,6 +1665,19 @@ bool OSDMonitor::check_failure(utime_t now, int target_osd, failure_info_t& fi)
     for (map<int,failure_reporter_t>::iterator p = fi.reporters.begin();
 	 p != fi.reporters.end();
 	 ++p) {
+      //get subtree reporter
+      if (reporter_subtree_level == "osd") {
+	reporters_by_subtree.insert("osd." + to_string(p->first));
+      } else {
+	map<string, string> reporter_loc = osdmap.crush->get_full_location(p->first);
+	map<string, string>::iterator iter = reporter_loc.find(reporter_subtree_level); 
+	if (iter != reporter_loc.end()) {
+	  reporters_by_subtree.insert(iter->second);
+	} else {
+	  reporters_by_subtree.insert("UNKNOWN");
+	}
+      }
```

Comment:
I wonder if we should flip this around so that the "osd" and unknown case are the same.  That way if someone misconfigures and specifies "foo" as the subtree level it will 'gracefully' degrade to N OSDs (instead of putting everything as unknown and never reaching the default threshold of 2).